Promise/Promise+Extras.swift

```diff
@@ -73,19 +99,30 @@ public enum Promises {
             }).then(fulfill).catch(reject)
         })
     }
-
+    
+    /// Runs a block of code that creates a `Promise` but which can also `throw`.
+    /// - Parameter block: Code to generate a `Promise`; `throw`s are rejected as expcted.
+    /// - Returns: `Promise<T>` that is created from the given block
     public static func kickoff<T>(_ block: @escaping () throws -> Promise<T>) -> Promise<T> {
         return Promise(value: ()).then(block)
     }
 
+    /// Allows for a block of code that could `throw` to be executed and turned into a `Promise`.
+    /// - Parameter block: Code to be executed; `throw`s are rejected as expected.
+    /// - Returns: `Promise<T>` that executes the given block, and rejects if necessary.
     public static func kickoff<T>(_ block: @escaping () throws -> T) -> Promise<T> {
         do {
             return try Promise(value: block())
         } catch {
             return Promise(error: error)
         }
     }
-
+    
+    /// Pairs two promises such that the results are paired together.
```

Comment:
```suggestion
    /// Pairs two promises and returns a promise with the results paired together.
    /// Fails if either fails.
```

Reply:
@khanlou This one was _way_ better; thanks. :beers: